test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -47,6 +58,17 @@ public void NonGeneric_GetRequiredService_Throws_WhenNoServiceRegistered()
         }
 
         [Fact]
+        public void IRequiredServiceAware_NonGeneric_GetRequiredService_Throws_WhenNoServiceRegistered()
+        {
+            // Arrange
+            var serviceProvider = new RequiredServiceAwareServiceProvider();
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredService(typeof(IFoo)),
```

Comment:
Wouldn't this just hit the RequiredServiceAwareServiceProvider.GetRequiredService method directly rather than the extension method given than it is using type-inference and the class exposes GetRequiredService as a public method?